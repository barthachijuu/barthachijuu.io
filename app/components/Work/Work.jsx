/**
 *
 * Work
 *
 * @version 1.0.0
 * @author Bartolo Amico
 * @since 16-Jul-2020
 */

import React from 'react';
import PropTypes from 'prop-types';
import { getDisplayDate } from 'Utility/utils';
import styles from './styles.js';


const Work = (props) => {
  const classes = styles();
  const viewJob = (entry, index, total) => (
    <div className="row item">
      <div className="twelve columns">
        <h3><a href={entry.website}>{entry.company}</a></h3>
        <p className={classes.info}>
          {entry.position}
          <span> &bull; </span>
          <span className={classes.infosummary}>{entry.summary}</span>
          <span> &bull; </span>
          <em className={classes.infosummary}>
            {getDisplayDate(entry.startDate, entry.endDate)}
          </em>
        </p>
        {entry.highlights.map(point => (
          <p className={classes.point}>
            <span className={classes.bulletpoint}>&bull; </span>
            {point}
          </p>
        ))}
      </div>
      {index === total ? <br /> : <hr />}
    </div>
  );

  return (
    <>
      <section id="work" className={classes.work}>
        <div className="row">
          <div className={`two columns ${classes.headercol}`}>
            <h1><span>Work</span></h1>
          </div>
          <div className={`ten columns ${classes.maincol}`}>
            {props.works.map((entry, index) => (viewJob(entry, index, props.works.length - 1)))}
          </div>
        </div>
      </section>
    </>
  );
};

Work.propTypes = {
  // Add here some propTypes
  // and set a minimal description i.e.
  /** Description of prop "works". */
  works: PropTypes.array,
};

Work.defaultProps = {
  // Add here some default propTypes values
  works: [],
};

export default Work;
