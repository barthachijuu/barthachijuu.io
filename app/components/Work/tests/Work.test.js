/**
 *
 * Jest Unit Test for
 * Work Component
 *
 */
import React from 'react';
import { shallow } from 'enzyme';
import Work from '../Work';

let component;
let baseProps;

beforeAll(() => {
  baseProps = {
    router: {
    },
  };
});

const works = [
  {
    company: 'Luna Labs',
    position: 'Software Analyst',
    website: 'https://https://www.lunalabs.it/',
    startDate: '2019-10-04',
    endDate: 'Present',
    summary: 'Milan, Italy',
    highlights: [
      'Analysis of the technical specifications of the projects and planning of development flows.',
      'Development of the source code and creation of the architectural structure.',
      'Supervision and maintenance of projects carried out by other teams.',
    ],
  },
  {
    company: 'Reply IrisCube',
    position: 'Senior Software Engineer (Team Lead)',
    website: 'https://www.reply.com/it/',
    startDate: '2015-06-08',
    endDate: 'Present',
    summary: 'Milan, Italy',
    highlights: [
      'He is developing part of the architectural section of an internal Intesa Sanpaolo project.',
      'Creation of the frontend architectural structure for Nexi\'s internal projects',
    ],
  },
];
beforeEach(() => {
  component = shallow(<Work {...baseProps} works={works} />);
});

describe('<Work />', () => {
  describe('Rendering', () => {
    it('should render self', () => {
      expect(component).toHaveLength(1);
    });
    it('should match the snapshot', () => {
      expect(component.html()).toMatchSnapshot();
    });
  });
});
