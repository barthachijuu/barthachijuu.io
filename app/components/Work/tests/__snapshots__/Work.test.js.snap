// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Work /> Rendering should match the snapshot 1`] = `"<section id=\\"work\\" class=\\"work-0-1-1\\"><div class=\\"row\\"><div class=\\"two columns headercol-0-1-3\\"><h1><span>Work</span></h1></div><div class=\\"ten columns maincol-0-1-2\\"><div class=\\"row item\\"><div class=\\"twelve columns\\"><h3><a href=\\"https://https://www.lunalabs.it/\\">Luna Labs</a></h3><p class=\\"info-0-1-4\\">Software Analyst<span> • </span><span class=\\"infosummary-0-1-5\\">Milan, Italy</span><span> • </span><em class=\\"infosummary-0-1-5\\">04-10-2019 - Present</em></p><p class=\\"point-0-1-6\\"><span class=\\"bulletpoint-0-1-7\\">• </span>Analysis of the technical specifications of the projects and planning of development flows.</p><p class=\\"point-0-1-6\\"><span class=\\"bulletpoint-0-1-7\\">• </span>Development of the source code and creation of the architectural structure.</p><p class=\\"point-0-1-6\\"><span class=\\"bulletpoint-0-1-7\\">• </span>Supervision and maintenance of projects carried out by other teams.</p></div><hr/></div><div class=\\"row item\\"><div class=\\"twelve columns\\"><h3><a href=\\"https://www.reply.com/it/\\">Reply IrisCube</a></h3><p class=\\"info-0-1-4\\">Senior Software Engineer (Team Lead)<span> • </span><span class=\\"infosummary-0-1-5\\">Milan, Italy</span><span> • </span><em class=\\"infosummary-0-1-5\\">08-06-2015 - Present</em></p><p class=\\"point-0-1-6\\"><span class=\\"bulletpoint-0-1-7\\">• </span>He is developing part of the architectural section of an internal Intesa Sanpaolo project.</p><p class=\\"point-0-1-6\\"><span class=\\"bulletpoint-0-1-7\\">• </span>Creation of the frontend architectural structure for Nexi&#x27;s internal projects</p></div><br/></div></div></div></section>"`;
